lib/linguist/heuristics.rb

```diff
@@ -160,6 +160,20 @@ def call(data)
       end
     end
 
+    disambiguate "M", "Mathematica", "Matlab", "Mercury", "Objective-C" do |data|
+      if /^[ \t]*@(interface|class|protocol|property|end|synchronised|selector|implementation)\b/.match(data)
+        Language["Objective-C"]
+      elsif data.include?(":- module")
```

Comment:
Whenever I've seen the sequence it's been at the start of a line but it's unusual enough that perhaps we don't need to be that strict?